indico/modules/events/registration/controllers/management/reglists.py

```diff
@@ -185,3 +204,108 @@ class RHRegistrationEdit(RHManageRegistrationBase):
     def _process(self):
         return WPManageRegistration.render_template('management/registration_modify.html', self.event,
                                                     registration=self.registration)
+
+
+class RHRegistrationsActionBase(RHManageRegFormBase):
+    """Base class for classes performing actions on registrations"""
+
+    def _checkParams(self, params):
+        RHManageRegFormBase._checkParams(self, params)
+        ids = set(request.form.getlist('registration_id'))
+        self.registrations = (Registration
+                              .find(Registration.id.in_(ids), ~Registration.is_deleted)
+                              .with_parent(self.regform)
+                              .all())
+
+
+class RHRegistrationEmailRegistrants(RHRegistrationsActionBase):
+    """Send email to selected registrants"""
+
+    def _checkParams(self, params):
+        self._doNotSanitizeFields.append('from_address')
+        RHRegistrationsActionBase._checkParams(self, params)
+
+    def _send_emails(self, form):
+        for registration in self.registrations:
+            email_body = replace_placeholders('registration-email', form.body.data, registration=registration)
+            template = get_template_module('events/registration/emails/custom_email.html',
+                                           email_subject=form.subject.data, email_body=email_body)
+            email = make_email(to_list=registration.email, cc_list=form.cc_addresses.data,
+                               from_address=form.from_address.data, template=template, html=True)
+            send_email(email, self.event, 'Registration')
+        return len(self.registrations)
+
+    def _process(self):
+        form = EmailRegistrantsForm()
+        if form.validate_on_submit():
+            num_emails_sent = self._send_emails(form)
+            flash(ngettext("The email was sent.",
+                           "{num} emails were sent", num_emails_sent).format(num=num_emails_sent), 'success')
+            return jsonify_data()
+        return WPManageRegistration.render_template('management/email.html', form=form)
+
+
+class RHRegistrationDelete(RHRegistrationsActionBase):
+    """Delete selected registrations"""
+
+    def _process(self):
+        for registration in self.registrations:
+            registration.is_deleted = True
+            logger.info('Registration {} deleted by {}'.format(registration, session.user))
+            # TODO: Signal for deletion?
+        num_reg_deleted = len(self.registrations)
+        flash(ngettext("Registration was deleted.",
+                       "{num} registrations were deleted", num_reg_deleted).format(num=num_reg_deleted), 'success')
+        return jsonify_data()
+
+
+class RHRegistrationCreate(RHManageRegFormBase):
+    """Create new registration (management area)"""
+
+    def _process(self):
+        form = make_registration_form(self.regform)()
```

Comment:
Nothing for now, but eventually it would be nice if we included administrative sections here. It's a manager doing it so it'd make sense to show them instead of having to edit the registration afterwards.